python/feldera/sql_context.py

```diff
@@ -269,34 +268,33 @@ def register_table_from_sql(self, ddl: str):
 
         self.tables[name] = SQLTable(name, ddl)
 
-    def connect_source_pandas(self, table_name: str, df: pandas.DataFrame):
+    def connect_source_pandas(self, table_name: str, df: pandas.DataFrame, flush: bool = False):
```

Comment:
In the hindsight, this design where we connect Pandas inputs to the pipeline before it's running was a mistake.  I suggest that we nuke the run_to_completion method. Instead the user must call `start`, then feed their dataframes and finally call `run_to_completion`. Feeding a dataframe to a pipeline that is not running is an error. Do you see any downsides to this approach?

Reply:
I think so as well, then there are the setup phase and the data phase of a pipeline. It also helps that then not everything needs to be loaded in advance, especially if it turns out that the pipeline does not start. Can't think of a particular downside, maybe losing some ability to schedule the different data to push? Like, one table-at-time, or all-table-in-parallel, it depends on the query which one makes sense.